logbook-spring-boot-starter/src/test/java/org/zalando/logbook/spring/ExcludeTest.java

```diff
@@ -0,0 +1,97 @@
+package org.zalando.logbook.spring;
+
+/*
+ * #%L
+ * Logbook: Spring
+ * %%
+ * Copyright (C) 2015 Zalando SE
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.TestPropertySource;
+import org.zalando.logbook.Correlator;
+import org.zalando.logbook.Logbook;
+import org.zalando.logbook.MockRawHttpRequest;
+import org.zalando.logbook.RawHttpRequest;
+
+import java.io.IOException;
+import java.util.Optional;
+import java.util.function.Predicate;
+
+import static java.util.Optional.empty;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@ContextConfiguration
+@TestPropertySource(properties = "spring.config.name = exclude")
+public final class ExcludeTest extends AbstractTest {
+
+    @Configuration
+    public static class TestConfiguration {
+
+        @Bean
+        public Logger httpLogger() {
+            final Logger logger = mock(Logger.class);
+            when(logger.isTraceEnabled()).thenReturn(true);
+            return logger;
+        }
+        
+        @Bean
+        public Predicate<RawHttpRequest> requestPredicate() {
+            return request -> 
+                    !"/health".equals(request.getRequestUri());
```

Comment:
> As @matosf correctly pointed out, real requests may be absolute, e.g. http://localhost/health.

That was my misunderstanding ;) Could still be done using a regex, though. But yes, it gets uglier.

> Implementor of what? The predicate or the RawHttpRequest interface?

The predicate.

> I liked the idea to use Ant-style path matchers, e.g. /admin/**. We could of course transform those into a regex, that might be less effort.

I'm afraid that I'm still missing something here. Are we just talking about the Spring Starter or Logbook in general? For the Spring Starter, using Ant matchers for the excludes list is probably the expected choice. So no need to add an additional custom request predicate that again just uses Ant matchers. Since it's already a Spring app, an additional Spring dependency should be no big deal.

Non-Spring apps have to create Logbook instances in a different way. It's just up to their decision how they implement the request predicate. Logbook should not bother, and currently, it indeed doesn't.

> That's what I would like to avoid. At least inside of Logbook. Could see this as its own project though.

Not sure why you want to avoid it, but this was just a thought. Perfectly fine with not having this inside Logbook as a util. If an app needs a special request predicate, it can come up best with it's own strategies how to match requests. If that involves more complex URL analysis, the app can do what fits best.